template-react-typescript/src/react-app-env.d.ts

```diff
@@ -0,0 +1,60 @@
+
+/// <reference types="react" />
+/// <reference types="react-dom" />
+
+declare namespace NodeJS {
+  interface ProcessEnv {
+    readonly NODE_ENV: 'development' | 'production' | 'test'
```

Comment:
The value of `process.env.NODE_ENV` is no longer fixed, it may be any string specified by the user using the `mode` option.